### PR TITLE
Lookup.resolveGroup returns an Iterator instead of List

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

**Comment by kokosing** [comment]
Why not `Stream`?

**Comment by sopel39** [comment]
> Why not Stream?

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Lookup.java [hunk 1]
```diff
@@ -55,18 +56,18 @@ static Lookup noLookup()
     {
         return node -> {
             verify(!(node instanceof GroupReference), "Unexpected GroupReference");
-            return ImmutableList.of(node);
+            return ImmutableList.of(node).iterator();
         };
     }
 
     static Lookup from(Function<GroupReference, List<PlanNode>> resolver)
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Lookup.java]: Shall you not update the `Function`'s type here too?

**Comment by kokosing** [comment]
Lets use `Stream`.

**Comment by sopel39** [comment]
> Iterator cannot be very easily wrapped within a Stream and it provides additional helper methods.

Yes, but it allows to write `for (Elem el: iterator)` loop, which is often useful. However, there is an `Stream::iterator()` method.

**Comment by sopel39** [comment]
applied comment

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Lookup.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
     @Deprecated
     default PlanNode resolve(PlanNode node)
     {
-        return getOnlyElement(resolveGroup(node));
+        return getOnlyElement(resolveGroup(node).iterator());
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Lookup.java]:
```
resolveGroup(node)
  .collect(toOptional())
  .get()
```
would be more javaeightish

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Lookup.java [hunk 3]
```diff
@@ -55,18 +55,18 @@ static Lookup noLookup()
     {
         return node -> {
             verify(!(node instanceof GroupReference), "Unexpected GroupReference");
-            return ImmutableList.of(node);
+            return ImmutableList.of(node).stream();
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Lookup.java]: `Stream.of(node)`

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Lookup.java [hunk 4]
```diff
@@ -55,18 +55,18 @@ static Lookup noLookup()
...
-    static Lookup from(Function<GroupReference, List<PlanNode>> resolver)
+    static Lookup from(Function<GroupReference, Stream<PlanNode>> resolver)
     {
         return node -> {
             if (node instanceof GroupReference) {
                 return resolver.apply((GroupReference) node);
             }
 
-            return ImmutableList.of(node);
+            return ImmutableList.of(node).stream();
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Lookup.java]: same

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Memo.java [hunk 5]
```diff
@@ -93,7 +93,7 @@ public PlanNode extract()
 
     private PlanNode extract(PlanNode node)
     {
-        return resolveGroupReferences(node, Lookup.from(planNode -> ImmutableList.of(this.resolve(planNode))));
+        return resolveGroupReferences(node, Lookup.from(planNode -> ImmutableList.of(this.resolve(planNode)).stream()));
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Memo.java]: `Stream.of(..)`

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java [hunk 6]
```diff
@@ -147,7 +147,7 @@ private RuleApplication applyRule()
     {
         SymbolAllocator symbolAllocator = new SymbolAllocator(symbols);
         Memo memo = new Memo(idAllocator, plan);
-        Lookup lookup = Lookup.from(planNode -> ImmutableList.of(memo.resolve(planNode)));
+        Lookup lookup = Lookup.from(planNode -> ImmutableList.of(memo.resolve(planNode)).stream());
```
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java]: same

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Lookup.java (hunk 2 above)
Comment (sopel39) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Lookup.java]: nice

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java [hunk 7]
```diff
@@ -76,7 +76,7 @@ public PlanNode optimize(PlanNode plan, Session session, Map<Symbol, Type> types
         }
 
         Memo memo = new Memo(idAllocator, plan);
-        Lookup lookup = Lookup.from(planNode -> ImmutableList.of(memo.resolve(planNode)));
+        Lookup lookup = Lookup.from(planNode -> ImmutableList.of(memo.resolve(planNode)).stream());
```
Comment (kokosing) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java]: No need to use list here. Please just use `Stream.of`

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java [hunk 8]
```diff
@@ -76,7 +76,7 @@ public PlanNode optimize(PlanNode plan, Session session, Map<Symbol, Type> types
         }
```
Comment (kokosing) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java]: Please also update commit message and PR title.

**Pull request closed without merging**